docs/md/expressions.md

```diff
@@ -1518,8 +1518,6 @@ Both text values are trimmed of surrounding whitespace and matching is case-inse
 @(has_pattern("Sell cheese please", "buy (\w+)")) → false
 @(has_pattern("Buy cheese please", "buy (\w+)")) → true
 @(has_pattern("Buy cheese please", "buy (\w+)").match) → Buy cheese
-@(has_pattern("Buy cheese please", "buy (\w+)").match.groups[0]) → Buy cheese
-@(has_pattern("Buy cheese please", "buy (\w+)").match.groups[1]) → cheese
```

Comment:
This is really not user-friendly as a way to get stuff because you have to repeat whatever expression you're using as the operand. I've removed it and think we need to come up with a better replacement for `@extra.0...1` which for now would be `@legacy_extra.0...1`

Reply:
Why remove the example though? Could still see people wanting to get at matches within an expression without having a totally separate split etc..

Reply:
We added a separate function for getting regex group matches, `regex_group(pattern, text, num)` - that's simpler than this `....match.groups[1]` syntax.

Reply:
Tho curiously I've just noticed we added it to the old engine but it seems I forgot to add it in the new engine